lib/cc/analyzer/container.rb

```diff
@@ -104,17 +102,22 @@ def read_stderr(err)
         end
       end
 
-      def timeout_thread(pid)
+      def timeout_thread
         Thread.new do
           sleep @timeout
           @timed_out = true
-          Process.kill("KILL", pid)
+          reap_running_container
         end
       end
 
       def container_data(duration: nil, status: nil)
         ContainerData.new(@image, @name, duration, status, @stderr_io.string)
       end
+
+      def reap_running_container
+        Analyzer.logger.warn("killing container name=#{@name}")
+        POSIX::Spawn::Child.new("docker", "kill", @name)
```

Comment:
I guess the line above would get logged and we could see if orphaned containers were attempted to be killed. It's probably fine. The only suggestion I would make is to execute this command in some way that raises if it fails (ie. backticks)